.github/workflows/precommit.yaml

```
@@ -2,35 +2,14 @@ name: Precommit tests
 
 on:
   push:
-    branches:
-      - '*'
+    branches: [main, develop, 127-fix-precommit-conditions]
   pull_request:
-    branches:
-      - '*'
+    branches: [main, develop]
 
 jobs:
   pre-commit:
-    needs: install-dev-tools
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
-      - name: dev env setup
-        uses: ./.github/actions/dev-env-setup
-      - name: set pre-commit cache directory
-        run: |
-          echo "PRE_COMMIT_HOME=$GITHUB_WORKSPACE/.pre-commit-cache" >> $GITHUB_ENV
-      - name: set PY
-        run: echo "PY=$(python -VV | sha256sum | cut -d' ' -f1)" >> $GITHUB_ENV
-      - name: create commitlint COMMIT_EDITMSG if not exists
-        run: |
-          if test -f ".git/COMMIT_EDITMSG"; then
-              echo "COMMIT_EDITMSG EXISTS, skipping"
-          else
-              touch .git/COMMIT_EDITMSG
-          fi
-      - uses: actions/cache@v3
-        with:
-          path: |
-            ./.pre-commit-cache
-          key: pre-commit-${{ env.PY }}-${{ hashFiles('.pre-commit-config.yaml') }}-v3
-      - run: pre-commit run --all-files
+      - uses: actions/setup-python@v3
```

```suggestion
      - uses: actions/setup-python@v5
```
This also needs a version for python